tiledb/sm/serialization/array_schema.cc

```diff
@@ -304,110 +305,175 @@ Status dimension_to_capnp(
   return Status::Ok();
 }
 
-Status dimension_from_capnp(
-    const capnp::Dimension::Reader& dimension_reader,
-    tdb_unique_ptr<Dimension>* dimension) {
-  Datatype dim_type = Datatype::ANY;
-  RETURN_NOT_OK(datatype_enum(dimension_reader.getType().cStr(), &dim_type));
-  dimension->reset(tdb_new(Dimension, dimension_reader.getName(), dim_type));
+/**
+ * Deserialize a tile extent from a cap'n proto object.
+ *
+ * Note: This API operates under the assumption that before its invocation,
+ * dimension_from_capnp has validated dim_type.
```

Comment:
doxygen has a section `@pre` to state preconditions, a standard way of saying "operates under the assumption" etc. Can just say `@pre dim_type is valid`. Then a comment is warranted in `dimension_from_capnp` at its validation call that it's doing it to satisfy the precondition of `tile_extent_from_capnp`.

Stating preconditions directly is better than referring to how a function is currently used in code.